routing/ROUTING_V1_HTTP.md

```diff
@@ -90,6 +90,18 @@ Response limit: 100 providers
 
 Each object in the `Providers` list is a *read provider record*.
 
+#### List of Routers
+
+A list of routers that should be consulted for responses. The specification imposes no constraints on the order by which the results are returned. Implementers are free to return results as they are found.
+
+If none are provided the backend may decide which ones to use. For a list of known routers see below.
+
+To select all routers use `?routers=*`. To determine the list of available routers see the `OPTIONS` API below
+
+### `OPTIONS /routing/v1/providers`
```

Comment:
Some quick thoughts: 
-  Learning about available routers via OPTIONS header introduces some UX challenges: 
   - can't paste link in address bar: need to pass custom method to `curl -X OPTIONS`, JS may need explicit entry in [Access-Control-Expose-Headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers) for the value to be readable via JS.
- Given the design decisions made when we abandoned Reframe and prioritized user-friendliness and human-readablinity of API, it may be a better  UX and less headache to support  listing available routers via `GET` and return them as JSON.

We can also have OPTIONS, but we should have GET.

#### Alternative idea for brainstorming:

Request: `GET /routing/v1/routers` (dedicated endpoint) or `GET /routing/v1/providers?routers=list`

Response: JSON with list of items following `router` schema

```json
{
  "Schema": "router",
  "Name": "ipfs-public-dht",
  "Default": false,
  "Features": ["providers", "ipns"],
  "Details": { .. } // optional opaque JSON with router-type-specific values
}
```